### DISCUSSION TITLE
Ice 0.11.0 (Beta 1)

### DISCUSSION THREAD

**Discussion by jordanbaird** [jordanbaird/Ice#333] [Announcements]
## What's New
* Search menu bar items using a custom hotkey
* Adjust the spacing between menu bar items
* Customize the time it takes to rehide temporarily shown menu bar items

## Known Issues
* Applying menu bar item spacing can occasionally fail
* Ice icon does not respect changes in menu bar item spacing

> [!NOTE]
> The above issues can be resolved by logging out and logging back in

## New Contributors
* @dhruvikn made their first contribution in https://github.com/jordanbaird/Ice/pull/277
* @stonerl made their first contribution in https://github.com/jordanbaird/Ice/pull/306

**Full Changelog**: https://github.com/jordanbaird/Ice/compare/0.10.5...0.11.0-beta.1

<hr /><em>This discussion was created from the release <a href='https://github.com/jordanbaird/Ice/releases/tag/0.11.0-beta.1'>Ice 0.11.0 (Beta 1)</a>.</em>

**Comment by Romachamp** [comment]
Hello! Menu bar search reveals an icon, if it is hidden, but it doesn’t hide it, when I close it. So, I have to manually drag it back into the hidden section. 

If the icon is visible, it changes it position, and doesn’t return to the original place as well. 

I would like just being able to open the icon, do something there, and then close it. So, it won’t change its position or move from the hidden section. However, I like the design of the search, and so far pretty good.

**Reply by jordanbaird** [reply]
It should automatically hide after a short period of time, which you can configure in the advanced settings pane. Does it not do that for you?

**Reply by Romachamp** [reply]
Well, it seems so. I waited about 5 seconds after closed icon menu, and it wasn’t hidden. Also, should it rearrange the icon after it is opened from the search bar? 

My MacOS version is Sonoma.

**Reply by jordanbaird** [reply]
The default delay is 15 seconds after showing the item. Then, if the item's menu is still open, it checks every 3 seconds until both the menu is closed and the application that owns the item is no longer active. This serves as kind of a "best guess" that you're done interacting with the item. Can you please wait a bit longer before moving it just to be sure? It could definitely be a bug, but I want confirmation before investigating it.

And yes, rearranging the item is intentional. It has to show the item to open its menu. But like I said, it should hide it again after your configured delay.

**Reply by Romachamp** [reply]
It hides it after 15 seconds, thanks. 

But, the issue with rearranging is connected, to the items, which aren't hidden in the menu bar manager, and which are always visible. When I select them in the menu bar search, Ice rearranges it, and doesn't return in place, even after 15 seconds. 

Also, it doesn't allow me to interact with system controls(toggle icon, which includes wifi, bluetooth, etc), when I select one in the menu bar search.

**Reply by jordanbaird** [reply]
Ah, I see. It probably shouldn't be moving items that are already visible, but that is in fact the current behavior. I'll make that change for the next beta. Not sure why it isn't returning to its original position...it does for me. Either way, it shouldn't be a problem anymore once it no longer moves.

In regard to the toggles, are you trying to use the keyboard to interact with them? That seems like a limitation of macOS. If I click the WiFi item, for example, then try to navigate to the toggle with the arrow keys, I can't. I'm not sure we'll be able to work around that.

**Reply by Romachamp** [reply]
By toggles I meant one icon with all in one, basically that icon, which includes wifi, Bluetooth, volume, etc. When I select it in the menu bar items search, it doesn’t open its menu, like with other items. Basically just closes menu bar search and doesn’t do anything with the item.

**Reply by jordanbaird** [reply]
Ah, I see what you mean. I'll get that fixed, too.
